xpublish_edr/formats/to_geotiff.py

```
@@ -0,0 +1,63 @@
+"""
+Generate GeoTIFF responses for an xarray dataset for EDR queries
+"""
+
+import xarray as xr
+from fastapi import Response
+
+
+def to_geotiff(ds: xr.Dataset) -> Response:
+    """Return a GeoTIFF response from an xarray dataset"""
+    import io
+
+    import numpy as np
+    import rasterio
+    from rasterio.transform import from_origin
+
+    # Get the first data variable
+    var_name = list(ds.data_vars)[0]
+    data = ds[var_name].values
+
+    # Handle different dimensions
+    if len(data.shape) == 2:
+        # 2D data (x,y)
+        data = np.expand_dims(data, axis=0)
+    elif len(data.shape) > 3:
+        # More than 3D, take first slice
+        data = data[0]
```

I'd rather just raise an error here.